results/pathway/hgsc/Project_node38genes_neg/js/xtrace_graph.js

```diff
@@ -0,0 +1,314 @@
+            })
+            
+            // Redraw the graph and such
+            dag.draw();
+        })
+        
+        function highlightPath(center) {        
+            var path = getEntirePathLinks(center);
+            
+            var pathnodes = {};
+            var pathlinks = {};
+            
+            path.forEach(function(p) {
+               pathnodes[p.source.id] = true;
+               pathnodes[p.target.id] = true;
+               pathlinks[p.source.id+p.target.id] = true;
+            });
+            
+            edges.classed("hovered", function(d) {
+                return pathlinks[d.source.id+d.target.id];            
+            })
+            nodes.classed("hovered", function(d) {
+                return pathnodes[d.id];
+            });
+            
+            var immediatenodes = {};
+            var immediatelinks = {};
+            immediatenodes[center.id] = true;
+            center.getVisibleParents().forEach(function(p) {
+                immediatenodes[p.id] = true;
+                immediatelinks[p.id+center.id] = true;
+            })
+            center.getVisibleChildren().forEach(function(p) {
+                immediatenodes[p.id] = true;
+                immediatelinks[center.id+p.id] = true;
+            })
+            
+            edges.classed("immediate", function(d) {
+                return immediatelinks[d.source.id+d.target.id];
+            })
+            nodes.classed("immediate", function(d) {
+                return immediatenodes[d.id];
+            })
+        }
+    }
+    
+    // The main draw function
+    this.draw = function() {
+        DAGTooltip.hide();                  // Hide any tooltips
+        console.log("draw begin")
+        var begin = (new Date()).getTime();  
+        var start = (new Date()).getTime();        
+        graphSVG.datum(graph).call(DAG);    // Draw a DAG at the graph attach
+        console.log("draw graph", new Date().getTime() - start);
+        start = (new Date()).getTime();    
+        minimapSVG.datum(graphSVG.node()).call(DAGMinimap);  // Draw a Minimap at the minimap attach
+        console.log("draw minimap", new Date().getTime() - start);
+        start = (new Date()).getTime();
+        graphSVG.selectAll(".node").call(DAGTooltip);        // Attach tooltips
+        console.log("draw tooltips", new Date().getTime() - start);
+        start = (new Date()).getTime();
+        setupEvents();                      // Set up the node selection events
+        console.log("draw events", new Date().getTime() - start);
+        start = (new Date()).getTime();
+        refreshViewport();                  // Update the viewport settings
+        console.log("draw viewport", new Date().getTime() - start);
+        start = (new Date()).getTime();
+        attachContextMenus();
+        console.log("draw contextmenus", new Date().getTime() - start);
+        console.log("draw complete, total time=", new Date().getTime() - begin);
```

Comment:
Just curious how long does it take to render?

Reply:
seconds. If you come by my desk, I can show you